okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.kt

```diff
@@ -566,5 +628,15 @@ class RealWebSocket(
      * the server doesn't respond the web socket will be canceled.
      */
     private const val CANCEL_AFTER_CLOSE_MILLIS = 60L * 1000
+
+    private const val HEADER_WS_EXTENSION = "Sec-WebSocket-Extensions"
+
+    /**
+     * Java Deflater does not allow specifying window size
+     * and the underlying zlib uses default value of MAX_WBITS=15.
+     * It's the default window size for many deflate implementations too.
```

Comment:
Oh sweet, we’re reading the same source code. I think we might get better interoperability by omitting the `server_max_window_bits` and `client_max_window_bits` parameters, since 15 bits is the default?

> Absence of this extension parameter in an extension negotiation
> response indicates that the server can receive messages compressed
> using an LZ77 sliding window of up to 32,768 bytes.

Reply:
Yeah, I think for correctness we actually need to remove these parameters:
https://tools.ietf.org/html/rfc7692#section-7.1.2.2

I think if we include a value, the server can agree on a lower value. But we don’t have a mechanism to use a smaller window bits size through the Java APIs. So our server would have say, a 4 KiB inflate window and the client would have a 32 KiB deflate window and we’d be in trouble.

Omitting the parameter altogether prevents the problem because the server is forbidden from returning a value for the parameter if the client doesn’t mention it.